src/metrics-gatherer.ts

```diff
@@ -134,81 +193,141 @@ export class MetricsGatherer {
 	public requestHandler(authTest?: AuthTestFunc): express.Handler {
-		return (req: express.Request, res: express.Response) => {
-			if (authTest && !authTest(req)) {
-				return res.status(403).send();
-			}
-			res.writeHead(200, { 'Content-Type': 'text/plain' });
-			res.end(prometheus.register.metrics());
+		let handler: express.Handler = (
+			_req: express.Request,
+			_res: express.Response,
+		) => {
+			console.error(
+				'error in metrics.requestHandler() caused empty express metrics request handler to be created',
+			);
 		};
+		try {
+			handler = (req: express.Request, res: express.Response) => {
+				if (authTest && !authTest(req)) {
+					return res.status(403).send();
+				}
+				res.writeHead(200, { 'Content-Type': 'text/plain' });
+				res.end(prometheus.register.metrics());
+			};
+		} catch (e) {
+			console.error('caught ' + e);
+			this.err(e);
+		}
+		return handler;
 	}
 
+	// create an express request handler given an auth test function which is
+	// suitable for use in a context where we're using node's `cluster` module
 	public aggregateRequestHandler(authTest?: AuthTestFunc): express.Handler {
-		const aggregatorRegistry = new prometheus.AggregatorRegistry();
-		return (req, res) => {
-			if (authTest && !authTest(req)) {
-				return res.status(403).send();
-			}
-			aggregatorRegistry
-				.clusterMetrics()
-				.then((metrics: string) => {
-					res.set('Content-Type', aggregatorRegistry.contentType);
-					res.send(metrics);
-				})
-				.catch((err: Error) => {
-					debug(`error in /cluster_metrics: ${err}`);
-					res.status(500).send();
-				});
+		let handler: express.Handler = (
+			_req: express.Request,
+			_res: express.Response,
+		) => {
+			console.error(
+				'error in metrics.aggregateRequestHandler() caused empty express metrics request handler to be created',
+			);
 		};
+		try {
+			const aggregatorRegistry = new prometheus.AggregatorRegistry();
+			handler = (req, res) => {
+				if (authTest && !authTest(req)) {
+					return res.status(403).send();
+				}
+				aggregatorRegistry
+					.clusterMetrics()
+					.then((metrics: string) => {
+						res.set('Content-Type', aggregatorRegistry.contentType);
+						res.send(metrics);
+					})
+					.catch((err: Error) => {
+						this.err(err);
+						res.status(500).send();
+					});
+			};
+		} catch (e) {
+			console.error('caught ' + e);
+			this.err(e);
+		}
+		return handler;
 	}
 
 	// get the prometheus output
 	public output(): string {
-		return prometheus.register.metrics();
+		let output = '';
+		try {
+			output = prometheus.register.metrics();
+		} catch (e) {
+			console.error('caught ' + e);
+			this.err(e);
+		}
+		return output;
 	}
 
 	// clear all metrics
 	public clear() {
-		prometheus.register.clear();
-		this.initState();
+		try {
+			prometheus.register.clear();
+			this.initState();
+		} catch (e) {
+			console.error('caught ' + e);
+			this.err(e);
+		}
+	}
+
+	private err(e: Error) {
+		debug(e);
```

Comment:
It's worth adding an informative string to this so there's a bit more context of where this comes from (not all errors include that themselves)

Reply:
specifically even knowing the module it's coming from can help a lot